fastjet/internal/delaunay/point.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/gonum/floats"
+)
+
+// Point in the X-Y Plane, that holds dynamic information about the
+// adjacent triangles, the nearest neighbor and the distance to that neighbor
+type Point struct {
+	// X and Y are the coordinates of the point.
+	X, Y float64
+	// adjacentTriangles is used for the triangulation, to find the nearest neighbor and
+	// to find the Voronoi Cell.
+	adjacentTriangles triangles
+	// isOutside indicates whether the point is outside the triangulation.
+	isOutside bool
+	// ID is a user defined identifier. There are no restrictions on how the user uses it.
```

Comment:
This still makes me very queasy. I don't like that a free user value is available that is used internally under strict invariants. Especially on a pointer value. This is how subtle bugs are born. Please find another way - at least discuss possible options that do not leave this open.

One approach is to keep ID only for internal use.